tools/labs/skels/kernel_api/6-list-sync/list-sync.c

```diff
@@ -68,11 +71,18 @@ static void task_info_add_to_list(int pid)
 		return;
```

Comment:
If you return here then you should also unlock the spinlock, otherwise it remains LOCKED and next time you call this function you will get a deadlock.